crates/cubecl-cpp/src/shared/element.rs

```diff
@@ -63,6 +124,13 @@ impl<D: Dialect> Elem<D> {
             Elem::_Dialect(_) => 0,
         }
     }
+
+    pub const fn size_bits(&self) -> usize {
+        match self {
+            Elem::FP4(_) => 4,
+            other => other.size() * 8,
+        }
+    }
```

Comment:
Is FP6 encoded on 1 full byte?

Reply:
yep, the top two bits are just `0`. Similar to `tf32`, except the conversion actually does zero out the bits while tf32 just ignores them (since `fp6` is actually a size change and `tf32` is just a reinterpet).